lib/guardian/topic_guardian.rb

```diff
@@ -199,6 +198,47 @@ def can_see_deleted_topics?(category)
     is_staff? || is_category_group_moderator?(category)
   end
 
+  # Accepts an array of `Topic#id` and returns an array of `Topic#id` which the user can see.
+  def can_see_topic_ids(topic_ids: [], hide_deleted: true)
```

Comment:
This is a tiny nit and you don't have to change it, but this method reads like a question to me (which of course the other guardian methods are) but we are filtering things out...something like `filter_inaccessible_topic_ids` might be better. But if you don't want to change it's totally fine, just a thought